File: Tests/EnsureTests.fs
namespace Extensions

open System
open Stringier.Extensions
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type EnsureTests() =
    [<TestMethod>]
    member _.``ensureBegins "Mr " "Bob Saget"`` () =
        Assert.AreEqual("Mr. Bob Saget", ensureBegins "Mr. " "Bob Saget")

    [<TestMethod>]
    member _.``ensureBegins "Mr " "Mr Bob Saget"`` () =
        Assert.AreEqual("Mr. Bob Saget", ensureBegins "Mr. " "Mr. Bob Saget")

    [<TestMethod>]
    member _.``ensureEnds " MD" "Gregory House"`` () =
        Assert.AreEqual("Gregory House MD", ensureEnds " MD" "Gregory House")

    [<TestMethod>]
    member _.``ensureEnds " MD" "Gregory House MD"`` () =
        Assert.AreEqual("Gregory House MD", ensureEnds " MD" "Gregory House MD")

File: Tests/EnsureTests.cs
using System;
using Stringier;
using Defender;
using Xunit;

namespace Tests {
	public class EnsureTests : Trial {
		[Theory]
		[InlineData("Bob Saget", "Mr. ", "Mr. Bob Saget")]
		[InlineData("Mr. Bob Saget", "Mr. ", "Mr. Bob Saget")]
		public void EnsureBeginsWith(String source, String required, String expected) => Claim.That(source.EnsureBeginsWith(required)).Equals(expected);

		[Theory]
		[InlineData("Gregory House", " MD", "Gregory House MD")]
		[InlineData("Gregory House MD", " MD", "Gregory House MD")]
		public void EnsureEndsWith(String source, String required, String expected) => Claim.That(source.EnsureEndsWith(required)).Equals(expected);
	}
}
